hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -509,6 +509,27 @@ private OmKeyInfo readKeyInfo(OmKeyArgs args, BucketLayout bucketLayout)
     if (args.getLatestVersionLocation()) {
       slimLocationVersion(value);
     }
+    Integer partNumberParam = args.getPartNumber();
+    if (partNumberParam != null && partNumberParam > 0) {
+      OmKeyLocationInfoGroup latestLocationVersion = value.getLatestVersionLocations();
+      if (latestLocationVersion != null && latestLocationVersion.isMultipartKey()) {
+
+        value.setKeyLocationVersions(
+            Collections.singletonList(
+                new OmKeyLocationInfoGroup(
+                    latestLocationVersion.getVersion(),
+                    value.getCurrentlocationsPartsMap()
+                        .getOrDefault(partNumberParam, Collections.emptyList()),
+                    true
+                )
+            )
+        );
+        value.setDataSize(
+            value.getCurrentDataSizePartsMap()
+                .getOrDefault(partNumberParam, 0L)
+        );
+      }
+    }
```

Comment:
I think we can just do main logic here. Just filter based location info based on the partNumber and calculate the data size based on the filtered part number.

Reply:
Main logic moved to this place